tasks/fullStack/app/controllers/reviews_controller.rb

```diff
@@ -1,44 +1,60 @@
 class ReviewsController < ApplicationController
+  protect_from_forgery except: :fetch_reviews
+
 
   DEFAULT_TAGS = ['default']
 
   def index
-    if params[:shop_id].present? && Shop.where("id = #{params[:shop_id]}").present?
-      params[:per_page] ||= 10
-      offset = params[:page].to_i * params[:per_page]
-
-      @data = []
-      products = Product.where("shop_id = #{params[:shop_id]}").sort_by(&:created_at)[offset..(offset + params[:per_page])]
-      products.each do |product|
-        reviews = product.reviews.sort_by(&:created_at)[offset..(offset + params[:per_page])]
-        @data << { product: product, reviews: reviews }
-      end
+    @products = Product.where(shop_id: params[:shop_id]).page params[:page]
```

Comment:
We shouldn't change the business logic code without the requirements or the automation tests

Reply:
Hey Lucas!
Yes, you are right, in my first approach I tried to handle everything without touching much the logic but I found myself within a problem when trying to paginate. For sure I wanted to remove that iteration and handle it at the db level but, to be honest, I just couldn't find a proper solution without breaking the pagination.